ocs_ci/deployment/deployment.py

```diff
@@ -0,0 +1,302 @@
+
+
+class Deployment(object):
+    """
+    Base for all deployment platforms
+    """
+    def __init__(self):
+        self.platform = config.ENV_DATA['platform']
+        self.ocp_deployment_type = config.ENV_DATA['deployment_type']
+        self.cluster_path = config.ENV_DATA['cluster_path']
+        self.deploy = config.RUN['cli_params']['deploy']
+        self.teardown = config.RUN['cli_params']['teardown']
+
+    def add_volume(self):
+        """
+        Implement add_volume in child class which is specific to
+        platform
+        """
+        raise NotImplementedError("add_volume functionality not implemented")
+
+    def deploy_cluster(self):
+        """
+        Implement deploy in child class
+        """
+        raise NotImplementedError("deploy functionality not implemented")
+
+    def deploy_ocp(self):
+        """
+        Implement ocp deploy in specific child class
+        """
+        raise NotImplementedError("deploy_ocp functionality not implemented")
+
+    def deploy_ocp_prereq(self):
+        """
+        Perform generic prereq before calling openshift-installer
+        This method performs all the basic steps necessary before invoking the
+        installer
+        """
+        # Test cluster access and if exist just skip the deployment.
+        if is_cluster_running(self.cluster_path):
+            logger.info(
+                "The OCP installation is skipped because the cluster is "
+                "running"
+            )
+            return False
+        elif self.teardown and not self.deploy:
+            logger.info(
+                f"Attempting teardown of non-accessible cluster: "
+                f"{self.cluster_path}"
+            )
+            return False
+        elif not self.deploy and not self.teardown:
+            msg = "The given cluster can not be connected to: {}. ".format(
+                self.cluster_path)
+            msg += (
+                f"Provide a valid --cluster-path or use --deploy to "
+                f"deploy a new cluster"
+            )
+            pytest.fail(msg)
+        elif not system.is_path_empty(self.cluster_path) and self.deploy:
+            msg = "The given cluster path is not empty: {}. ".format(
+                self.cluster_path)
+            msg += (
+                f"Provide an empty --cluster-path and --deploy to deploy "
+                f"a new cluster"
+            )
+            pytest.fail(msg)
+        else:
+            logger.info(
+                f"A testing cluster will be deployed and cluster information "
+                f"stored at: %s",
+                self.cluster_path
+            )
+
+        # Generate install-config from template
+        logger.info("Generating install-config")
+        pull_secret_path = os.path.join(
+            constants.TOP_DIR,
+            "data",
+            "pull-secret"
+        )
+
+        # TODO: check for supported platform and raise the exception if not
```

Comment:
Those comments can be deleted and you should implement this in get_deployment method from factory and raise some specific exception like DeploymentPlatformNotSupported

Reply:
Done, Introduced new exception 'DeploymentPlatformNotSupported'